### PR TITLE
Fault detection udpates

### PR DESCRIPTION
**Pull request by Mnoble-19**
**_WHAT DOES THIS PR DO?_**

- [x] Improves fault detection job with additional approach (isolation forest)
- [x] Small changes to  packaging & dependency installation process for airflow
Screenshot 2024-03-17 at 14 20 01 (2)

**_HOW DO I TEST OUT THIS PR?_**
- [x] Workflows: [Link to Workflows README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/workflows#readme)

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] {{baseUrl}}/api/v2/predict/faulty-devices - @OchiengPaul442

### CODE REVIEW

File: src/workflows/Dockerfile [hunk 1]
```diff
@@ -3,6 +3,7 @@ RUN pip install --upgrade pip setuptools wheel
 WORKDIR airqo_etl_utils
 COPY airqo_etl_utils/* airqo_etl_utils/
 RUN python airqo_etl_utils/setup.py bdist_wheel
+#TODO: Review packaging process for this microservice
```
Comment (Baalmart) [inline src/workflows/Dockerfile:6]: Thanks! Is this code comment still needed in our codebase? @Mnoble-19

**Review by Baalmart — COMMENTED** [review]
All looks good to me. Will wait for @BenjaminSsempala 's review and then proceed with any necessary actions. Thanks @Mnoble-19 !

File: src/workflows/Dockerfile (hunk 1 above)
Comment (Mnoble-19) [reply src/workflows/Dockerfile:6]: Yes @Baalmart , I've tried to make some changes but not yet done. 
package management in python is weird and my knowledge is also still shaky for now thus the TODO, but by end of next sprint I'll probably have read more on it and will remove it

Comment (Baalmart) [reply src/workflows/Dockerfile:6]: Okay @Mnoble-19 , I was told that you are on leave so we shall leave your PR open until you return. I also encourage @BenjaminSsempala to review it asap.

**Review by BenjaminSsempala — APPROVED** [review]

**Pull request merged**